rfcs/rfc-0000-pronouns.md

```diff
@@ -0,0 +1,87 @@
+- Feature Name: Pronouns
+- Start Date: (16/06/2023)
+- RFC PR: None
+- Tracking Issue: None
+- Status: draft
+
+# Summary
+
+This change creates a pronoun field for Revolt.chat, this field will be attached to a users public profile information.
+
+
+# Motivation
+
+Pronouns have been requested by the community and are high up on the RFC list.
+
+# Guide-level explanation
+
+Currently users display their pronouns in their Profile content (bio) but this provides a cleaner solution to the problem. The aim of this change is to ensure that pronouns for a user can be easily seen and displayed inline with a users displayname / username.
+
+# Reference-level explanation
+
+The field `pronouns` will be added to the main `User` object, as well as the `PartialUser` and `FieldsUser` objects. (In practise this means that the pronouns field will be fetched for the `fetch_user` API route, and pronouns can be edited or removed from the `edit_user` API route.
+
+The pronoun field (on a `User` will be an optional array of strings. 
+
+### Data representation in Rustlang
+```rust
+{
+pronouns: Option<Vec<String>>
+}
+```
+
+### Pronouns stored in JSON
+```json
+{
+	"pronouns": ["she", "her"]
+}
+```
+For no pronouns the object will not be serialised 
+```json
+
+```
+
+Additionally, there will be database changes for pronouns to be fully implemented. The `Collection` shall contain an entry of every supported language (represented by an abbreviation) with an array of every valid pronoun for the language. This collection must also be accessible as an API route.
+
+```bson
+{
+  "language": "<string>",
+  "pronouns": ["<string>", "<string>", ...]
+}
+```
```

Comment:
For what should this be exactly? Set a list of pronouns to choose from based on the language you use?
Isn't that a bit limiting? Would apreciate some clarification here.

Reply:
The purpose for the collection is to stop trolls from using the feature to enter joke pronouns, as for language I've decided that it'll start with English only.